crates/extensions/tedge_config_manager/src/actor.rs

```diff
@@ -548,11 +550,11 @@ impl ConfigOperation {
             Ok(None)
         } else if config.config_snapshot_topic.accept(message) {
             Ok(Some(ConfigOperation::Snapshot(
-                ConfigSnapshotCmdPayload::from_json(message.payload_str()?)?,
+                ConfigSnapshotCmd::payload_from(message.payload_bytes())?,
             )))
         } else if config.config_update_topic.accept(message) {
             Ok(Some(ConfigOperation::Update(
-                ConfigUpdateCmdPayload::from_json(message.payload_str()?)?,
+                ConfigUpdateCmd::payload_from(message.payload_bytes())?,
```

Comment:
Beware that this usage of `payload_from` will be made useless by https://github.com/thin-edge/thin-edge.io/pull/2844 which purpose is to use the same pattern for Log&Config management as for Software management: i.e. to have the log manager MQTT independent. The `request_from_message` method will be removed.